### ISSUE TITLE
modularizeImports not fully work

### ISSUE THREAD

**Issue by bluelovers** [vercel/next.js#44474]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

.

### Link to the code that reproduces this issue

https://stackblitz.com/edit/github-zr8cps?file=pages/index.tsx&view=editor

### To Reproduce

https://stackblitz.com/edit/github-zr8cps?file=pages/index.tsx&view=editor

### Describe the Bug

modularizeImports not fully work

```js
const nextConfig = {
  modularizeImports: {
    '../components/halves': {
      transform: '../components/halves/{{ member }}',
    },
    lodash: {
      transform: 'lodash-es/{{member}}',
    },
    'lodash\\.([_\\w]+)': {
      transform: 'lodash-es/{{ matches.[1] }}',
    },
    'lodash/((([_\\w])?\/?)*)': {
      transform: 'lodash-es/{{ matches.[1] }}/{{member}}',
    },
  },
}
```

```jsx
// @ts-ignore
import { LeftHalf, RightHalf } from '../components/halves'
import _defaults from 'lodash.defaults';
import { merge, debounce } from 'lodash';
import { eq } from 'lodash-es';
// @ts-ignore
import _deburrLetter from 'lodash/_deburrLetter';

export default function Index() {
  _defaults;
  merge;
  eq;
  debounce;
  _deburrLetter;
  return (
    <div>
      <LeftHalf />
      <RightHalf />
    </div>
  )
}
```

### Expected Behavior

![image](https://user-images.githubusercontent.com/167966/210158424-1f197eef-c4e6-4169-a019-5c5c60212d3a.png)

**Comment by altechzilla** [comment]
Yup, same issue over here. Trying to import `@mui/icons-material` from a transpiled module doesn't recognize the components properly. For example, when importing `import { Send } from '@mui/icons-material';` and then rendering `<Send />`, it will throw an error. Only works if you do `<Send.default />`.

**Comment by camiblanch** [comment]
I have been experiencing the same problem as the previous commenter. I did some digging and checked to see how things work in `create-next-app`. I created a fresh, new project using `create-next-app` then added `@mui/icons-material`. I was able to see the icon working there. I noticed in my project in the `package.json` file I have `"type": "module"` and the new project does not. If I set the `type` in the new project, I get the same error message as in my project. Is there a way to work around that? I need to be using  `"type": "module"`

**Comment by hongdeyuan** [comment]
I have been experiencing the same problem；Is there any solution？
